src/ucp/core/ucp_request.c

```diff
@@ -564,20 +520,19 @@ UCS_PROFILE_FUNC_VOID(ucp_request_memory_dereg, (context, datatype, state, req),
     ucs_trace_func("context=%p datatype=0x%"PRIx64" state=%p", context,
                    datatype, state);
 
-    if (req->flags & UCP_REQUEST_FLAG_USER_MEMH) {
-        return;
-    }
-
     switch (datatype & UCP_DATATYPE_CLASS_MASK) {
     case UCP_DATATYPE_CONTIG:
-        ucp_request_dt_dereg(context, &state->dt.contig, 1, req);
+        if (state->dt.contig.memh != NULL) {
+            ucp_request_dt_dereg(context, &state->dt.contig.memh, 1, req);
```

Comment:
maybe ucp_request_dt_dereg can handle NULL memh (ignore) ?

Reply:
it's a double pointer, so need to check for NULL for both memhs and all memhs[i], kinda awkward?

Reply:
yes, just in ucp_request_dt_dereg

Reply:
we call ucp_request_dt_dereg four times and only in one case we check memh for NULL, so I think it's better to do it outside in this specific case

Reply:
ok